File: 0024.两两交换链表中的节点/0024-两两交换链表中的节点.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if (!head || !head->next)
            return head;
        ListNode n_head(0);
        ListNode* pre = &n_head;
        n_head.next = head;
        ListNode* ptr = head;
        while(ptr && ptr->next){
            pre->next = ptr->next;
            // if (!ptr->next)
            ptr->next = ptr->next->next;
            // else
                // ptr->next = NULL;
            pre->next->next = ptr;
            pre = pre->next->next;
            ptr = ptr->next;
        }
        return n_head.next;
        
    }
};

File: 0024.两两交换链表中的节点/0024-两两交换链表中的节点.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
            return head
        dummy = ListNode(1)
        dummy.next = head

        first = head
        second = head.next
        
        tail = second.next
        first.next = self.swapPairs(tail)
        second.next = first
        dummy.next = second
        
        return dummy.next